### ISSUE TITLE
Turning axes_style and plotting_context into contextdecorators

### ISSUE THREAD

**Issue by anntzer** [mwaskom/seaborn#774]
Python3's `contextmanager` decorator actually creates contextdecorators, i.e. they can be used as decorators to apply themselves to the whole decorated function.
I believe this same functionality can be easily added to seaborn's `axes_style` and `plotting_context` simply by adding a `__call__` method to the `_AxesStyle` and `_PlottingContext` classes (and would be useful to have).
Thoughts?

**Comment by mwaskom** [comment]
So the idea would be that you would have in your codebase something like

```python
@sns.axes_style("ticks")
def my_plot(ax, ...):
    ...
```

and then any time you call `my_plot` it will make a plot with the `ticks` style?

**Comment by anntzer** [comment]
Yes.

**Comment by mwaskom** [comment]
Sounds cool! What does the `__call__` method have to do? Do you want to put together a PR?

**Comment by anntzer** [comment]
Something like (untested)
```
def __call__(self, func):
    @functools.wraps(func)
    def wrapper(*args, **kwargs):
        with self: return func(*args, **kwargs)
    return wrapper
```
I have a few other things on my plate right now so the PR will have to wait if you want me to do it :-)

**Comment by mwaskom** [comment]
Closed with #775
